src/status_im2/contexts/wallet/receive/view.cljs

```diff
@@ -0,0 +1,102 @@
+(ns status-im2.contexts.wallet.receive.view
+  (:require
+    [quo.core :as quo]
+    [react-native.blur :as blur]
+    [react-native.core :as rn]
+    [react-native.platform :as platform]
+    [react-native.safe-area :as safe-area]
+    [react-native.share :as share]
+    [reagent.core :as reagent]
+    [status-im2.common.qr-codes.view :as qr-codes]
+    [status-im2.constants :as constants]
+    [status-im2.contexts.wallet.common.sheets.network-preferences.view :as network-preferences]
+    [status-im2.contexts.wallet.receive.style :as style]
+    [utils.i18n :as i18n]
+    [utils.image-server :as image-server]
+    [utils.re-frame :as rf]))
+
+(def id-to-network
+  {constants/mainnet-chain-id  :ethereum
+   constants/optimism-chain-id :optimism
+   constants/arbitrum-chain-id :arbitrum})
+
+(defn- share-action
+  [address share-title]
+  (share/open
+   (if platform/ios?
+     {:activityItemSources [{:placeholderItem {:type    "text"
+                                               :content address}
+                             :item            {:default {:type "text"
+                                                         :content
+                                                         address}}
+                             :linkMetadata    {:title share-title}}]}
+     {:title   share-title
+      :subject share-title
+      :message address})))
+
+(defn- open-preferences
+  [selected-networks]
+  (rf/dispatch [:show-bottom-sheet
+                {:content
+                 (fn []
+                   [network-preferences/view
+                    {:blur?             true
+                     :selected-networks (set @selected-networks)
+                     :on-save           (fn [chain-ids]
+                                          (rf/dispatch [:hide-bottom-sheet])
+                                          (reset! selected-networks (map #(get id-to-network %)
+                                                                         chain-ids))
+                                        )}])}]))
+
+
+(defn view
+  []
+  (let [padding-top       (:top (safe-area/get-insets))
+        wallet-type       (reagent/atom :wallet-legacy)
+        selected-networks (reagent/atom [:ethereum :optimism :arbitrum])]
+    (fn []
+      (let [{:keys [address color emoji] :as account} (rf/sub [:wallet/current-viewing-account])
+            share-title                               (str (:name account) " " (i18n/label :t/address))
+            qr-url                                    (if (= @wallet-type :wallet-multichain)
```

Comment:
we have wallet qr's on many pages. imo we should move this function to a utility ns. 
additionally it would be nice to have a simple test covering the logic

Reply:
@J-Son89 done!